datafusion/core/src/datasource/physical_plan/parquet/arrow_statistics.rs

```diff
@@ -0,0 +1,252 @@
+    if !matches!(arrow_datatype, DataType::Int64) {
+        return Err(DataFusionError::Internal(format!(
+            "Unsupported data type {:?} for statistics",
+            arrow_datatype
+        )));
+    }
+
+    // row counts
+    let row_count = statistics
+        .rg_statistics
+        .iter()
+        .map(|rg| rg.row_count)
+        .collect::<Vec<_>>();
+
+    // get null counts
+    let parquet_stats = statistics.rg_statistics.iter().map(|rg| rg.statistics);
+    let null_counts = parquet_stats
+        // .map(|stats| stats.and_then(|s| Some(s.null_count())))
+        .map(|stats| stats.map(|s| s.null_count()))
+        .collect::<Vec<_>>();
+
+    // get min and max values
+    let parquet_stats_with_min_max = statistics
+        .rg_statistics
+        .iter()
+        .map(|rg| rg.get_statistics())
+        .collect::<Vec<_>>();
+
+    let mins = parquet_stats_with_min_max
+        .iter()
+        .map(|stats| {
+            stats.and_then(|s| {
+                let stats = ParquetColumnRowGroupStatistics::try_as_i64(s)?;
+                Some(*stats.min())
+            })
+        })
+        .collect::<Vec<_>>();
+
+    let maxs = parquet_stats_with_min_max
+        .iter()
+        .map(|stats| {
+            stats.and_then(|s| {
+                let stats = ParquetColumnRowGroupStatistics::try_as_i64(s)?;
+                Some(*stats.max())
+            })
+        })
+        .collect::<Vec<_>>();
+
+    Ok(ArrowStatistics {
+        min: Arc::new(Int64Array::from(mins)),
+        max: Arc::new(Int64Array::from(maxs)),
+        row_count: Arc::new(UInt64Array::from(row_count)),
+        null_count: Arc::new(UInt64Array::from(null_counts)),
+    })
+}
+
+/// All row group statistics of a file for a column
+pub struct ParquetColumnStatistics<'a> {
+    // todo: do we need this?
+    // arrow column schema
+    // column_schema: &'a FieldRef,
+    _column_name: &'a str, // todo: do we need this?
+    rg_statistics: Vec<ParquetColumnRowGroupStatistics<'a>>,
+}
+
+/// Row group statistics of a column
+pub struct ParquetColumnRowGroupStatistics<'a> {
+    row_count: u64,
+    statistics: Option<&'a ParquetStatistics>,
+}
+
+impl<'a> ParquetColumnRowGroupStatistics<'a> {
+    /// Create a new instance of `ParquetColumnRowGroupStatistics`
+    pub fn new(row_count: u64, statistics: Option<&'a ParquetStatistics>) -> Self {
+        Self {
+            row_count,
+            statistics,
+        }
+    }
+
+    /// Return statistics if it exists and has min max
+    /// Otherwise return None
+    pub fn get_statistics(&self) -> Option<&'a ParquetStatistics> {
+        let stats = self.statistics?;
+        if stats.has_min_max_set() {
+            Some(stats)
+        } else {
+            None
+        }
+    }
+
+    /// Return the statistics as ValuesStatistcs<i64> if the column is i64
+    /// Otherwise return None
+    fn try_as_i64(stats: &'a ParquetStatistics) -> Option<&'a ValueStatistics<i64>> {
```

Comment:
For this PR, we focus on Int64 only. This function will be refactored in  following PR(s)